### ISSUE TITLE
spring-cloud-consul-config:1.0.0.M1 depends on a snapshot build that is not available

### ISSUE THREAD

**Issue by tehjakey** [spring-cloud/spring-cloud-consul#72]
I'm not able to resolve two transient dependency of cloud-consul-config. I believe this is because they are snapshot builds rather than released artifacts. I've fixed the problem locally, but it's a hack.
Here's the fix:
```
compile ('org.springframework.cloud:spring-cloud-consul-config:1.0.0.M1'){
        exclude group: 'com.ecwid.consul', module: 'consul-api'
        exclude group: 'org.springframework.cloud', module: 'spring-cloud-commons'
        exclude group: 'org.springframework.cloud', module: 'spring-cloud-context'
        }
    compile 'com.ecwid.consul:consul-api:1.1.4'
    compile 'org.springframework.cloud:spring-cloud-commons:1.0.2.RELEASE'
    compile 'org.springframework.cloud:spring-cloud-context:1.0.2.RELEASE'
```
Here's the problem. See the failed dependency resolution.
```
+--- org.springframework.cloud:spring-cloud-consul-config:1.0.0.M1
|    +--- org.springframework.cloud:spring-cloud-consul-core:1.0.0.M1
|    |    +--- org.springframework.boot:spring-boot-starter-web:1.2.3.RELEASE (*)
|    |    +--- org.springframework.boot:spring-boot-starter-actuator:1.2.3.RELEASE
|    |    |    +--- org.springframework.boot:spring-boot-starter:1.2.3.RELEASE (*)
|    |    |    +--- org.springframework.boot:spring-boot-actuator:1.2.3.RELEASE (*)
|    |    |    \--- org.springframework:spring-core:4.1.6.RELEASE (*)
|    |    +--- org.springframework.cloud:spring-cloud-commons:1.0.2.BUILD-SNAPSHOT FAILED
|    |    \--- com.google.code.gson:gson:2.3.1
|    \--- org.springframework.cloud:spring-cloud-context:1.0.2.BUILD-SNAPSHOT FAILED
```

**Comment by spencergibb** [comment]
I'm aware of the issue of have the milestone depend on a snapshot, [see my note](https://github.com/spring-cloud/spring-cloud-consul/blob/1.0.0.M1/pom.xml#L56).  What I'm less sure of is why you can't see them.  They are in the [snapshot repo](http://repo.spring.io/snapshot/org/springframework/cloud/spring-cloud-commons/1.0.2.BUILD-SNAPSHOT/).  Do you have that as a repo?  1.0.0.M2 will fix this problem.

**Comment by tehjakey** [comment]
I can access the snapshot repo by adding that repo in the build script, but we are using artifactory as our only "prescribed" repo... and artifactory is not configured to allow access to the snapshot. I'm told opening artifactory to pull in snapshot builds is not desirable, so I can use the workaround above and wait for the 1.0.0.M2 build. Thank you for getting back to me so quickly

**Comment by bitzl** [comment]
The reference of a Snapshot-build breaks our build process, too. Any plans when 1.0.0.M2 might be ready?

**Comment by spencergibb** [comment]
@bitzl before spring one (Sept 14, 2015)

**Comment by spencergibb** [comment]
Closed via fe866b732c3dda153f99d242631403951b6e85b6
